utils/logger.js

```diff
@@ -0,0 +1,107 @@
+const LogLevel = {
+  LOG: {
+    value: 0,
+    method: console.log,
+  },
+  INFO: {
+    value: 1,
+    method: console.info,
+  },
+  DEBUG: {
+    value: 2,
+    method: console.debug,
+  },
+  WARN: {
+    value: 3,
+    method: console.warn,
+  },
+  ERROR: {
+    value: 4,
+    method: console.error,
+  },
+};
+
+class Logger {
+  constructor(scope, level) {
+    this.level = +level || LogLevel.ERROR.value;
+    this.scope = scope || "";
+  }
+
+  setLogLevel(levelStr) {
+    this.level = levelStr
+      ? LogLevel[levelStr.toString().toUpperCase()].value
+      : this.level;
```

Comment:
In case user provides something that is not included in LogLevel array will throw an error while accessing the `.value`. Handle that scenario.

Reply:
Good catch. Fixed it.